shared/src/price_estimate.rs

```diff
@@ -61,32 +98,39 @@ impl PriceEstimating for UniswapPriceEstimator {
         buy_token: H160,
         amount: U256,
         kind: OrderKind,
-    ) -> Result<f64> {
+    ) -> Result<BigRational> {
         if sell_token == buy_token {
-            return Ok(1.0);
+            return Ok(num::one());
         }
         if amount.is_zero() {
             return self
                 .best_execution_spot_price(sell_token, buy_token)
                 .await
-                .and_then(|(_, price)| {
-                    big_rational_to_float(price)
-                        .ok_or_else(|| anyhow!("Cannot convert price ratio to float"))
-                });
+                .map(|(_, price)| price);
         }
-
         match kind {
             OrderKind::Buy => {
                 let (_, sell_amount) = self
                     .best_execution_buy_order(sell_token, buy_token, amount)
                     .await?;
-                Ok(sell_amount.to_f64_lossy() / amount.to_f64_lossy())
+                Ok(BigRational::new(
```

Comment:
nanonit: Maybe a comment that amount is zero is already checked above? Otherwise it couldn't hurt to handle this case anyways (in case we remove the 0 handling and forget to update this line).

Reply:
Oops sorry, I haven't caught up with this repo reviewing dynamics :) I'll include the comment as part of the next PR.